modules/loaders-common/src/main/scala/com/snowplowanalytics/snowplow/loaders/transform/NonAtomicFields.scala

```diff
@@ -47,11 +47,26 @@ object NonAtomicFields {
     failure: FailureDetails.LoaderIgluError
   )
 
+  sealed trait ResolveTypesException extends Throwable
```

Comment:
I'm curious, what is the reason for the new trait?  Why not `case class ClashingFieldNames(...) extends Throwable`?

Reply:
Just wanted to make it more general in case we want to add other cases later on. I don't have strong opinion, I can remove it if you want.